### DISCUSSION TITLE
Use Pillow for graphic overlays in addition to text?

### DISCUSSION THREAD

**Discussion by keresian** [aaronwmorris/indi-allsky#1611] [Ideas]
To preface this: Just trying to see if any thought has been given to the possibility.   For instance: Overlay a graphical representation of the globe with daylight cycle as the day progresses.  Or Moon phase as a graphic, or "live weather radar" image. Pillow has the support to overlay one image on another, but from what I can tell, its not currently implemented.

If it isnt already planned, I may work to add the functionality myself.

**Comment by aaronwmorris** [comment]
Overlaying graphics is not difficult, even without Pillow.  It is fairly easy to overlay images with pure numpy code.  Is there a common place to have some of these images generated?

**Reply by keresian** [reply]
I assume a few of the images could be static.  like moon phases, since you already get the label, you could just have 10 static moon images (each 10% coverage).  I think the challenging ones would be sun rise/set on a globe (since you might want it centered on your location, but once generated it could easily be static.. maybe as part of the setup you run a script that generates it as one time thing), and weather (because most APIs cost money)

**Reply by aaronwmorris** [reply]
I am going to work on a dynamically generated projection of the moon.  I'm not sure about the others without a source of data.

**Comment by aaronwmorris** [comment]
Merged #1614 with the Moon Overlay functionality.  You can find the overlay settings on the Image tab under the Image Label settings (closer to the bottom).

The overlay is shaded for the real-time accurate illumination of the moon.

**Reply by keresian** [reply]
This works great.  I assume if there was a way to get a "globe" of a persons current location generally, the same math could be applied base on sun position to move 'nighttime' across the image for day/night cycle?

**Reply by aaronwmorris** [reply]
I assume you mean something like https://www.timeanddate.com/worldclock/sunearth.html .  I would love to implement something like this, but this would probably require a bit of time to understand the math involved.

**Comment by keresian** [comment]
I was thinking something even easier.  You know the sun angle and how fast it moves.. so if I live in indiana, either the system or I pull a one time image from google map at a certain zoom rate so you know the size of the image mile/km wise, and then you could use the angle to sweep a 'dark' area over the area, then when the sun is close to coming back up, sweep the dark image off.  You could even cheese it since the only times the image needs to change are the hour or so around dusk and dawn, and you could even pre-generate those overlay images and just swap them in (so like dawn-5degree.png would be what it looks like at -5 degrees sun angle, etc).

Your answer would be infinitely more awesome, but also infinitely more difficult.

**Comment by keresian** [comment]
You could use this for weather:  openstreetmap is free for all, as is rainviewer.

https://www.rainviewer.com/api/weather-maps-api.html
openstreetmaps for the roads, rainviewer for the radar.  complete with example api's.
